a10_octavia/network/drivers/neutron/a10_octavia_neutron.py

```diff
@@ -12,697 +12,169 @@
-            port['port']['fixed_ips'] = [fixed_ip]
+    def create_port(self, network_id, subnet_id=None, fixed_ip=None):
+        new_port = None
+        if not subnet_id:
+            subnet_id = self.neutron_client.get_network(network_id).subnets[0]
+        fixed_ip = {'subnet_id': subnet_id}
         try:
+            port = {'port': {'name': 'octavia-port-' + network_id,
+                             'network_id': network_id,
+                             'admin_state_up': True,
+                             'device_owner': OCTAVIA_OWNER,
+                             'fixed_ips': [fixed_ip]}}
+            if fixed_ip:
+                port['port']['fixed_ips'][0]['ip_address'] = fixed_ip
             new_port = self.neutron_client.create_port(port)
-        except Exception as e:
-            message = _('Error creating neutron port on network '
-                        '{network_id}.').format(
-                network_id=load_balancer.vip.network_id)
+            new_port = utils.convert_port_dict_to_model(new_port)
+        except Exception:
+            message = "Error creating port in network: {0}".format(network_id)
             LOG.exception(message)
-            raise base.AllocateVIPException(
-                message,
-                orig_msg=getattr(e, 'message', None),
-                orig_code=getattr(e, 'status_code', None),
-            )
-        new_port = utils.convert_port_dict_to_model(new_port)
-        return self._port_to_vip(new_port, load_balancer)
+            raise exceptions.PortCreationFailedException(message)
+        return new_port
 
-    def unplug_aap_port(self, vip, amphora, subnet):
-        interface = self._get_plugged_interface(
-            amphora.compute_id, subnet.network_id, amphora.lb_network_ip)
-        if not interface:
-            # Thought about raising PluggedVIPNotFound exception but
-            # then that wouldn't evaluate all amphorae, so just continue
-            LOG.debug('Cannot get amphora %s interface, skipped',
-                      amphora.compute_id)
-            return
+    def delete_port(self, port_id):
         try:
-            self.unplug_network(amphora.compute_id, subnet.network_id)
+            self.neutron_client.delete_port(port_id)
         except Exception:
-            pass
-        try:
-            aap_update = {'port': {
-                'allowed_address_pairs': []
-            }}
-            self.neutron_client.update_port(interface.port_id,
-                                            aap_update)
-        except Exception:
-            message = _('Error unplugging VIP. Could not clear '
-                        'allowed address pairs from port '
-                        '{port_id}.').format(port_id=vip.port_id)
+            message = "Error deleting port: {0}".format(port_id)
             LOG.exception(message)
-            raise base.UnplugVIPException(message)
 
-        # Delete the VRRP port if we created it
+    def get_port_by_ip(self, ip):
```

Comment:
This isn't required. The only reason this was done was because the `create_ve_port` was not returning the port it was created.